packages/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js

```diff
@@ -0,0 +1,897 @@
+/* eslint no-use-before-define: 0 */
+import React from 'react';
+import ReactDOM from 'react-dom';
+// eslint-disable-next-line import/no-unresolved
+import ReactDOMServer from 'react-dom/server';
+// eslint-disable-next-line import/no-unresolved
+import ShallowRenderer from 'react-test-renderer/shallow';
+// eslint-disable-next-line import/no-unresolved
+import TestUtils from 'react-dom/test-utils';
+import checkPropTypes from 'prop-types/checkPropTypes';
+import has from 'has';
+import {
+  ConcurrentMode,
+  ContextConsumer,
+  ContextProvider,
+  Element,
+  ForwardRef,
+  Fragment,
+  isContextConsumer,
+  isContextProvider,
+  isElement,
+  isForwardRef,
+  isPortal,
+  isSuspense,
+  isValidElementType,
+  Lazy,
+  Memo,
+  Portal,
+  Profiler,
+  StrictMode,
+  Suspense,
+} from 'react-is';
+import { EnzymeAdapter } from 'enzyme';
+import { typeOfNode } from 'enzyme/build/Utils';
+import shallowEqual from 'enzyme-shallow-equal';
+import {
+  displayNameOfNode,
+  elementToTree as utilElementToTree,
+  nodeTypeFromType as utilNodeTypeFromType,
+  mapNativeEventNames,
+  propFromEvent,
+  assertDomAvailable,
+  withSetStateAllowed,
+  createRenderWrapper,
+  createMountWrapper,
+  propsWithKeysAndRef,
+  ensureKeyOrUndefined,
+  simulateError,
+  wrap,
+  getMaskedContext,
+  getComponentStack,
+  RootFinder,
+  getNodeFromRootFinder,
+  wrapWithWrappingComponent,
+  getWrappingComponentMountRenderer,
+  compareNodeTypeOf,
+} from 'enzyme-adapter-utils';
+import findCurrentFiberUsingSlowPath from './findCurrentFiberUsingSlowPath';
+import detectFiberTags from './detectFiberTags';
+
+// Lazily populated if DOM is available.
+let FiberTags = null;
+
+function nodeAndSiblingsArray(nodeWithSibling) {
+  const array = [];
+  let node = nodeWithSibling;
+  while (node != null) {
+    array.push(node);
+    node = node.sibling;
+  }
+  return array;
+}
+
+function flatten(arr) {
+  const result = [];
+  const stack = [{ i: 0, array: arr }];
+  while (stack.length) {
+    const n = stack.pop();
+    while (n.i < n.array.length) {
+      const el = n.array[n.i];
+      n.i += 1;
+      if (Array.isArray(el)) {
+        stack.push(n);
+        stack.push({ i: 0, array: el });
+        break;
+      }
+      result.push(el);
+    }
+  }
+  return result;
+}
+
+function nodeTypeFromType(type) {
+  if (type === Portal) {
+    return 'portal';
+  }
+
+  return utilNodeTypeFromType(type);
+}
+
+function isMemo(type) {
+  return compareNodeTypeOf(type, Memo);
+}
+
+function isLazy(type) {
+  return compareNodeTypeOf(type, Lazy);
+}
+
+function unmemoType(type) {
+  return isMemo(type) ? type.type : type;
+}
+
+function transformSuspense(renderedEl, prerenderEl, { suspenseFallback }) {
```

Comment:
If I remove this method and all associated code, it goes from 5 suspense related test failures to 16 suspense related test failures. This code is clearly doing something useful.